src/Infrastructure/Elastic/ElasticIndexAdapter.php

```diff
@@ -35,4 +71,44 @@ public function flush(string $index): void
             'body' => $matchAllQuery
         ]);
     }
+
+    private function upsertAlias(IndexAliasConfigurationInterface $aliasConfiguration): void
+    {
+        $exists = $this->client->indices()->existsAlias(['name' => $aliasConfiguration->getAliasName()]);
+
+        if (!$exists) {
+            $this->client->indices()->putAlias([
+                'index' => $aliasConfiguration->getIndexName(),
+                'name' => $aliasConfiguration->getAliasName(),
+            ]);
+        } else {
+            $this->client->indices()->updateAliases([
+                'body' => [
+                    'actions' => [
+                        ['add' => ['index' => $aliasConfiguration->getAliasName() . '*', 'alias' => $aliasConfiguration->getAliasName().'-history']],
+                        ['remove' => ['index' => '*', 'alias' => $aliasConfiguration->getAliasName()]],
+                        ['add' => ['index' => $aliasConfiguration->getIndexName(), 'alias' => $aliasConfiguration->getAliasName()]],
```

Comment:
I don't understand what these 3 lines do. Maybe explain it in a comment, or create a function for it with an explaining name.